src/Compilers/Helpers/AnalyzerAssemblyLoadUtils.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class AnalyzerAssemblyLoadUtils
+    {
+        public static AssemblyIdentity TryGetAssemblyIdentity(string filePath)
+        {
+            try
+            {
+                if (!File.Exists(filePath))
```

Comment:
We should remove this check.  The `FileStream` constructor should essentially do it for us.